libtsuba/include/tsuba/AsyncOpGroup.h

```
@@ -3,24 +3,27 @@
 
 #include <future>
 #include <list>
+#include <optional>
 
 #include "katana/Result.h"
 
 namespace tsuba {
 
+using AsyncError = std::optional<katana::CopyableErrorInfo>;
```

I'm wondering if we should just have a `CopyableResult` which is like a `Result` but uses a string for error info and we can remove `CopyableErrorInfo` since it hasn't really caught on.

---

Using only a string as the error info would discard the error code making selective error handling impossible. So I would prefer creating `CopyableResult` which uses `CopyableErrorInfo`.

---

yes, sorry; that is what I should have said

---

Sounds good. I'll aim to do that instead.

---

It's less trivial than I thought at the beginning. This is going to take me some time since I have other things going on.